src/app/icd/server/ICDMonitoringTable.cpp

```diff
@@ -242,15 +259,21 @@ CHIP_ERROR ICDMonitoringTable::Get(uint16_t index, ICDMonitoringEntry & entry) c
 
 CHIP_ERROR ICDMonitoringTable::Find(NodeId id, ICDMonitoringEntry & entry)
 {
-    uint16_t index = 0;
-    while (index < this->Limit())
+    CHIP_ERROR err;
+    uint16_t index;
+    ICDMonitoringEntry tempEntry(mSymmetricKeystore);
+
+    for (index = 0; index < this->Limit(); index++)
     {
-        ReturnErrorOnFailure(this->Get(index++, entry));
-        if (id == entry.checkInNodeID)
+        SuccessOrExit(err = this->Get(index, tempEntry));
```

Comment:
I'm not sure i understand why the PSA key slots would not be clearer by this implementation compared of the previous one.

Reply:
here we first load only to `tempEntry`. and then we have two options:
- entry is found, all data from `tempEntry` is copied to `entry`. Then if we call `CreateKey` it will be cleared (the same behavior as before)
- entry is not found, only `entry.index` is updated upon `exit:` . This way `CreateKey` will call clearing a default handler, which is zero-initialized (and that handle is always an invalid handle for PSA). Before, `entry` would keep the key handles of last loaded entry (because of `ReturnErrorOnFailure(this->Get(index++, entry));`). So when we called key creation on that, it would call clearing an already existing slot from a different entry.